meetings/2024-12-11.md

```diff
@@ -0,0 +1,134 @@
+- MdJ: the main question: how the data moves
+  - ... who is sending data, who is receiving data
+  - ... it doesn't matter the exact route the data takes
+  - ... let's decide on "ClientClient" spec?
+- eP: it can be confusing, very similar C2C or CC specs.
+- MdJ: "client interop" specs?
+- eP: "apps and pods"?
+- MdJ: what about SAI ? (joke)
+    - ... by example the [chat spec](https://github.com/solid/chat) is a good example of "client client spec".
+- eP. and we havent said anything about "domain" yet
+  - ... knowledge domain? domain interop? 
+- H: "data domain specs"
+- eP: data domain specs. but no concsensus or contract as it blends with other concepts
+  - ... Thanks, Niko, i'll take it from here.
+- eP: Data Domain Specs seems neutral, probably the best choice
+- MdJ: Maybe we should talk to Tim
+
+### Trusted applications
+* MdJ: The 'trusted applications' mechanism we implemented in [NSS v5](https://github.com/nodeSolidServer/node-solid-server/blob/v5.0.0/CHANGELOG.md#500) was always meant to be a stop-gap. As solidcommunity.net is switching from NSS to Pivot, we need something to replace it. We are not yet at the point where working implementations of SAI exist. Therefore, I'm thinking of adding an app launcher along the lines of https://solid.pondersource.com/, that [edits the WAC ACLs and type index files](https://github.com/inrupt/launcher-exploration/blob/master/src/services/preparePod/preparePodForApp.ts) to allow apps as you launch them.
+  - ... we don't have SAI working in CSS.
+* ep: How do you identify apps in WAC ACLs?
+* MdJ: Web apps use the WebID of the logged in user and the client id is the app URL.
+  - ... the WAC spec says that that should work
+  - ... I don't know how it's implemented in specific servers.
+  - ... The special thing use is that the client ID matches the domain of the redirect URL from OAuth.
+* eP: If that that uses ACL orgin but not as the orgin header, I think it's not currently defined that way in WAC.
+  - ... In Solid OIDC we have to 2 options currently. [...]
+  - ... There is no global ID identifier such as WebID. 
+  - ... We can pick it up at one of the next meetings.
+* MdJ: we should check the WAC spec to make sure it's clear. I don't think we should move away from dynamic registrations. That's because you can trust a random app, even if the server never saw that app before.
+* eP: if you use URLs as client IDs, you don't need to do dynamic registrations. https://datatracker.ietf.org/doc/draft-parecki-oauth-client-id-metadata-document/
+* MdJ: so it's dynamic, although it doesn't use the OIDC dynamic registration.
+  - ... I opened a PR on pivot to change a dialog, related to this conversation here.
+  - ... [from chat]: https://github.com/solid-contrib/pivot/pull/38
+* eP: I don't think r/w that data, but all the data you have access to. If you're in a social graph, interacting with other users, that's a problem.
```

Comment:
What does `r/w` mean, here? (See the last line.) The usual `read/write` doesn't make sense in this context.
```suggestion
* MdJ: We should check the WAC spec to make sure it's clear. I don't think we should move away from dynamic registrations, because you might trust a random app, even if the server never saw that app before.
* eP: If you use URLs as client IDs, you don't need to do dynamic registrations. https://datatracker.ietf.org/doc/draft-parecki-oauth-client-id-metadata-document/
* MdJ: So it's dynamic, although it doesn't use the OIDC dynamic registration.
  - ... I opened a PR on Pivot to change a dialog, related to this conversation here.
  - ... [from chat]: https://github.com/solid-contrib/pivot/pull/38
* eP: I don't think r/w that data, but all the data you have access to. If you're in a social graph, interacting with other users, that's a problem.
```